spaghetto/src/Spago/Cmd.js

```diff
@@ -1,48 +0,0 @@
-import { execa, execaCommand } from "execa";
-
-const mkOptions = (opts) => {
-  let options = { preferLocal: true, shell: opts.shell };
-  if (opts.cwd) {
-    options.cwd = opts.cwd;
-  }
-  if (opts.input) {
-    options.input = opts.input;
-  }
-  return options;
-};
-
-export function spawnImpl(cmd, args, opts) {
-  return execa(cmd, args, mkOptions(opts));
-}
-
-export function spawnCommandImpl(cmd, opts) {
-  return execaCommand(cmd, mkOptions(opts));
-};
-
-export function pipeStdinImpl(cp) {
-  cp.stdin.pipe(process.stdin);
-}
-
-export function pipeStdoutImpl(cp) {
-  cp.stdout.pipe(process.stdout);
-}
-export function pipeStderrImpl(cp) {
-  cp.stderr.pipe(process.stderr);
-}
-
-export function joinImpl(cp, onError, onSuccess) {
-  return cp.then((data) => onSuccess(data))
-    .catch((err) => {
-      // WTFFFF, this is because on ENOENT we don't get an exitCode
-      if (err.exitCode === undefined) {
-        err.exitCode = -1;
-      }
```

Comment:
What specifically triggered this case? I need to see if `node-execa` handles it properly.

Reply:
This is because your implementation has a `Maybe ExitCode`, where you get `Nothing` if the executable you're trying to call does not exist.
The idea is that I always wanted an exit code returned to avoid having the application layer deal with a `Nothing`, but I think it's fine to have a `Maybe` in there as long as it's documented.

Reply:
Where should this be documented?

Reply:
In `execa`, no need to document it here